### DISCUSSION TITLE
Multisite Environment: incompatibility with Elementor / Elementor Pro any version

### DISCUSSION THREAD

**Discussion by devTTF** [stuttter/ludicrousdb#151] [Q&A]
*** EDITED ***

LUDICROUS DB is working but system finds incompatibility with Elementor Pro

```
Fatal error: Uncaught Exception: Table 'wp_113_e_events' already exists
in /home/customer/www/dashboard.thetraderforge.com/public_html/dashttf/wp-content/plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php on line 1525

Call stack:

mysqli_query()
wp-content/plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php:1525
LudicrousDB::_do_query()
wp-content/plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php:1398
LudicrousDB::query()
wp-content/plugins/elementor/core/common/modules/event-tracker/db.php:152
Elementor\C\C\M\E\DB::create_table()
wp-content/plugins/elementor/core/common/modules/event-tracker/db.php:184
Elementor\C\C\M\E\DB::__construct()
wp-content/plugins/elementor/core/common/modules/event-tracker/module.php:40
Elementor\C\C\M\E\Module::__construct()
wp-content/plugins/elementor/core/common/app.php:71
Elementor\C\C\App::init_components()
wp-content/plugins/elementor/includes/plugin.php:767
Elementor\Plugin::init_common()
wp-content/plugins/elementor/includes/frontend.php:231
Elementor\Frontend::init()
wp-includes/class-wp-hook.php:324
WP_Hook::apply_filters()
wp-includes/class-wp-hook.php:348
WP_Hook::do_action()
wp-includes/plugin.php:517
do_action()
wp-includes/template-loader.php:13
require_once()
wp-blog-header.php:19
require()
index.php:17
```

Tried to delete table, but when the system recreates it, fatal error occurs

*** Previously... *

Hi Ludicrous Community!
I need some help.

I'm trying to configure LudicrousDB so that tables created with NS Cloner (now WP Clone) for the subsites I'm interested in end up in another database.

Let me explain further:

I have a subsite that I clone each time so that everything I need is already set up. Since I have many subsites and each one has a specific number in the table prefix, I want to move certain subsites to another database (for example, those containing numbers from 51 to 100 in the database prefix).

The cloning process creates them in the first database. The simplest thing for me is to export those tables, once created, to the new database and make sure LudicrousDB understands where they ended up.

I've set everything up as per the guide, but I find that LudicrousDB always reads from the first database.
If I delete the tables from the first database, it returns 'not found', so there's definitely something that I'm missing in all of this.

Do you have any advice on how to proceed?

I modified the db-config.php file as follows:

```
/**
 * This is the most basic way to add a server to LudicrousDB using only the
 * required parameters: host, user, password, name.
 * This adds the DB defined in wp-config.php as a read/write server for
 * the 'global' dataset. (Every table is in 'global' by default.)
 */
$wpdb->add_database( array(
    'host'     => DB_HOST,     // If port is other than 3306, use host:port.
    'user'     => DB_USER,
    'password' => DB_PASSWORD,
    'name'     => DB_NAME,
    'dataset'  => 'global',
) );

/**
 * This adds the same server again, only this time it is configured as a slave.
 * The last three parameters are set to the defaults but are shown for clarity.
 */
 
$wpdb->add_database( array(
    'host'     => 'localhost',     // If port is other than 3306, use host:port.
    'user'     => 'user01',
    'password' => 'pwd01',
    'name'     => 'db02',
    'dataset' =>  'sites_from_51_to_100'
) );

$wpdb->add_database( array(
    'host'     => 'localhost',     // If port is other than 3306, use host:port.
    'user'     => 'user01',
    'password' => 'pwd01',
    'name'     => 'db03,
    'dataset' =>  'sites_from_101_to_150'
) );

$wpdb->add_callback('resolve_with_dataset');

function resolve_with_dataset($query, $wpdb) {
    if ( preg_match("/^{$wpdb->base_prefix}(5[1-9]|[6-9][0-9]|100)_/i", $wpdb->table) ) {
        // code to execute if the pattern matches
        return 'sites_from_51_to_100';
    } elseif ( preg_match("/^{$wpdb->base_prefix}(10[1-9]|11[0-9]|120|130|140|150)_/i", $wpdb->table) ) {
        //  code to execute if the pattern matches numbers between 101 and 150
        return 'sites_from_101_to_150';
    } else {
        return null; // Default to global dataset if no match
    }
}
```

Thank you in advance for your valuable assistance.

**Comment by JJJ** [comment]
That error tells us MySQL received a `CREATE TABLE` for a table that already exists, but the old stack trace does not tell us why the existence check and the create query disagreed.

One thing I would clean up first is the nested `plugins/ludicrousdb/ludicrousdb/` path shown in the trace. Keep one current copy of LudicrousDB, then install its drop-ins from there. The range callback also needs a small correction if it is still in use: `(10[1-9]|1[1-4][0-9]|150)` covers every site from 101 through 150.

If this still happens with current Elementor, WordPress, and LudicrousDB releases, could you post those exact versions, the complete `CREATE TABLE` query from the error, and a sanitized current `db-config.php`? Please also say which physical database already contains `wp_113_e_events` and which dataset the callback is expected to return for it. That should let us distinguish a routing mismatch from an old Elementor upgrade path. Be sure to remove hosts, usernames, passwords, and database names before posting.
